mlir/include/mlir/Dialect/SCF/Transforms/TileUsingInterface.h

```diff
@@ -190,10 +190,14 @@ tileAndFuseProducerOfSlice(RewriterBase &rewriter,
 /// where `%0` had other uses as well. If not reconstructed from within the loop
 /// body, uses of `%0` could not be replaced, making it still live and the
 /// fusion immaterial.
+///
+/// The @param `yieldResultNumber` decides which result would be yield. If not
+/// given, yield all `opResult` of fused producer.
```

Comment:
Could you make this more clear. I always find `std::optional<ArrayRef<unsigned>>` to be a bit ambiguous. What does it mean if we pass an `ArrayRef<unsigned>{}`?

Reply:
Good question, I understand what your concern. IMO, whether yield replacement for fused producer or not is controlled by `fusionControlFn` rather than this function. If user does not want to yield replacement for fused producer, this function should never be called, Otherwise, it expects for the non-empty `yieldResultNumber` and that is why `std::nullopt` represents all results here. As for `ArrayRef<unsigned>{}`, here is some possible solution:
1. throw failure.
2. treat it as same as `std::nullopt`. 
3. use `ArraryRef{}` as default argument, then no ambiguity between `std::nullopt` and `ArraryRef{}`.

Let me know your thought.

Reply:
> 1. throw failure.
> 2. treat it as same as std::nullopt.
> 3. use ArraryRef{} as default argument, then no ambiguity between std::nullopt and ArraryRef{}

Select the third one as current solution.

Reply:
I think (3) make sense, and it should just early exit on empty?

Reply:
So if this is empty, then that defaults to all the results being returned?

Reply:
Yes, exactly. Please see code [here](https://github.com/llvm/llvm-project/pull/93144/files#diff-088c31bdb684f78d48f7607d0007daa4e1fba53b0361def9841f79631d8a3d5dR953-R956).